tensorboard/plugins/profile/pod_viewer/pod_viewer_dashboard/pod-viewer-dashboard.ts

```diff
@@ -0,0 +1,235 @@
+  is: 'pod-viewer-dashboard',
+  properties: {
+    /**
+     * @type {?podviewer.proto.PodViewerInputData}
+     */
+    data: {
+      type: Object,
+      observer: '_dataChanged',
+    },
+    /**
+     * Active elements selected to be shown in the details card.
+     */
+    activeDetails: {
+      type: Array,
+      notify: true,
+    },
+    selectedChannel: {
+      type: Array,
+      notify: true,
+      observer: '_selectedChannelChanged',
+    },
+    activeBar: {
+      type: Object,
+      notify: true,
+      observer: '_activeBarChanged',
+    },
+    curStepId: {
+      type: Number,
+      value: 0,
+    },
+    _podStatsMaps: {
+      type: Object,
+      computed: '_computePodStatsMaps(data)',
+    },
+    _maxStepId: {
+      type: Number,
+      computed: '_computeMaxStepId(_podStatsMaps)',
+    },
+    _error_message: {
+      type: String,
+      computed: '_computeErrorMessage(_maxStepId)',
+    },
+    _runEnvironment: {
+      type: Object,
+      computed: '_computeRunEnvironment(data)',
+    },
+    _stepBreakdownLayers: {
+      type: Array,
+      value: () => [
+        {key: 'highFlopsComputeUs', label: 'High flops compute'},
+        {key: 'lowFlopsComputeUs', label: 'Low flops compute'},
+        {key: 'hostInfeedDurationUs', label: 'Infeed'},
+        {key: 'hostOutfeedDurationUs', label: 'Outfeed'},
+        {key: 'crsDurationUs', label: 'All reduce'},
+        {key: 'sendDurationUs', label: 'Send'},
+        {key: 'recvDurationUs', label: 'Recv'},
+      ],
+    },
+    _podStatsMap: {
+      type: Object,
+      computed:
+          '_computePodStatsMap(_podStatsMaps, curStepId, _stepBreakdownLayers)',
+    },
+    _stepStats: {
+      type: Array,
+      value: null,
+      computed: '_computeStepStats(_podStatsMap)',
+    },
+    _channelDb: {
+      type: Array,
+      value: null,
+      computed: '_computeChannelDb(_podStatsMap)',
+    },
+    _allReduceDb: {
+      type: Array,
+      value: null,
+      computed: '_computeAllReduceDb(_podStatsMap)',
+    },
+    _channelLayers: {
+      type: Array,
+      value: () => [
+        {key: 'durationUs', label: 'Duration (s)'},
+      ],
+    },
+    _allReduceLayers: {
+      type: Array,
+      value: () => [
+        {key: 'durationUs', label: 'Duration (µs)'},
+      ],
+    },
+    _stepBreakdownFunc: {
+      type: Array,
+      value: () => (d) => `(${d.chipId}, ${d.nodeId})`,
+    },
+    _channelFunc: {
+      type: Object,
+      value: () => (d) => d.channelId,
+    },
+    _allReduceFunc: {
+      type: Object,
+      value: () => function(d) {
+               if (!d.name) return;
+               const res =
+                   d.name.replace(/ll-reduce.|usion.|ll-reduce|usion/, '');
+               return res.length > 1 ? res : res + '0';
+             },
+    },
+  },
+  _computePodStatsMaps(data: podviewer.proto.PodViewerInputData|undefined|null):
+                       Array<podviewer.proto.PodStatsMap> {
+    if (!data) return;
```

Comment:
IMO, it would make sense to return an empty array here instead of returning an undefined. WDYT?

This applies to most `_compute*` methods.

Reply:
Changed this function to return [].
For channelDb, allReduceDb and stepStats,  need to return undefined for the following to work.
<stack-bar-chart hidden=[[!channelDb]]>